src/translator/byte_array_util.cpp

```diff
@@ -0,0 +1,33 @@
+#include "byte_array_util.h"
+#include <stdlib.h>
+#include <iostream>
+
+namespace marian {
+namespace bergamot {
+
+AlignedMemory loadFileToMemory(const std::string& path, size_t alignment){
+  uint64_t fileSize = filesystem::fileSize(path);
+  io::InputFileStream in(path);
+  ABORT_IF(in.bad(), "Failed opening file stream: {}", path);
+  AlignedMemory alignedMemory(fileSize, alignment);
+  in.read(reinterpret_cast<char *>(alignedMemory.begin()), fileSize);
+  ABORT_IF(alignedMemory.size() != fileSize, "Error reading file {}", path);
+  return alignedMemory;
+}
```

Comment:
This is now passed around as return (is return copy or by reference). I'm always a bit scared and I write 
```cpp
void loadFileToMemory(const std::string& path, size_t alignment, AlignedMemory &out)
```

Is ownership proper here, since `AlignedMemory` I'm guessing behaves similar to an `std::string`?

Reply:
https://stackoverflow.com/questions/4986673/c11-rvalues-and-move-semantics-confusion-return-statement